### PR TITLE
=reception immediately flush already known regs to a listing

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/apple/swift-distributed-actors/issues/1048

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-server-bot test this please

File: Sources/DistributedActors/Cluster/Reception/OperationLogDistributedReceptionist.swift [hunk 1]
```diff
@@ -553,21 +559,20 @@ extension OpLogDistributedReceptionist {
...
             }
 
-        case .remove(let key, let identity):
-            //            let resolved = system._resolveUntyped(context: .init(id: id, system: system))
+        case .remove(let anyKey, let identity):
             let resolved = try! actorSystem._resolveStub(identity: identity) // TODO(distributed): remove the throwing here?
 
-            unwatch(resolved)
-            if self.storage.removeRegistration(key: key, guest: resolved) != nil {
+            unwatchTermination(of: resolved)
```
Comment (ktoso) [inline Sources/DistributedActors/Cluster/Reception/OperationLogDistributedReceptionist.swift:574]: (unwatch is deprecated)

File: Sources/DistributedActors/Receptionist/DistributedReceptionist.swift [hunk 2]
```diff
@@ -100,13 +100,16 @@ extension DistributedReception {
                 self.underlying = AsyncStream<Element> { continuation in
                     let anySubscribe = AnyDistributedReceptionListingSubscription(
                         subscriptionID: ObjectIdentifier(self),
+                        file: file, line: line,
```
Comment (ktoso) [inline Sources/DistributedActors/Receptionist/DistributedReceptionist.swift:103]: this way we get even more precise diagnosis about "where the subscription was made" when we debug

File: Sources/DistributedActors/Receptionist/DistributedReceptionist.swift [hunk 3]
```diff
@@ -353,6 +356,11 @@ internal final class AnyDistributedReceptionListingSubscription: Hashable, @unch
     let subscriptionID: ObjectIdentifier
     let key: AnyDistributedReceptionKey
 
+    #if DEBUG
+    let file: String
+    let line: UInt
+    #endif
```
Comment (ktoso) [inline Sources/DistributedActors/Receptionist/DistributedReceptionist.swift:362]: Figured maybe just in debug mode... we could make it always any time hm

File: Sources/DistributedActors/utils.swift [hunk 4]
```diff
@@ -99,6 +110,17 @@ func pprint(_ message: StaticString, file: String = #filePath, line: UInt = #lin
...
+func pprint(_ message: StaticString, metadata: [String: Any], file: String = #filePath, line: UInt = #line) {
+    print("""
+    [pprint]\
+    [\(_createTimeFormatter().string(from: Date()))] \
+    [\(file):\(line)]\
+    {\(metadata.map { "\($0):\($1)" }.joined(separator: ";"))}
+    [\(_hackyPThreadThreadId())]: \
+    \(message)
+    """)
+}
```
Comment (ktoso) [inline Sources/DistributedActors/utils.swift:122]: not used in commit; but I use those println debugging a lot when debugging issues, so adding them

File: Sources/DistributedActors/Receptionist/DistributedReception.swift [hunk 5]
```diff
@@ -29,7 +29,6 @@ extension DistributedReception {
     ///
     /// The id defaults to "*" which can be used "all actors of that type" (if and only if they registered using this key,
     /// actors which do not opt-into discovery by registering themselves WILL NOT be discovered using this, or any other, key).
-    // FIXME(distributed): __DistributedClusterActor must go away, we don't need to be aware of `Message`
```
Comment (yim-lee) [inline Sources/DistributedActors/Receptionist/DistributedReception.swift:32]: 👏

**Review by yim-lee — APPROVED** [review]

**Pull request merged**